### ISSUE TITLE
训练后的模型求显著性图每次不一样

### ISSUE THREAD

**Issue by ivylinden** [thuml/Autoformer#166]
你好，我用autoformer训练完成后，对测试集反向求显著性图，发现每次的导数不一样，我已经确认模型是eval模式，没有其他随机值，想问下是否网络中有随机的部分

**Comment by wuhaixu2016** [comment]
您好，我这边尝试模型没有出现这个问题，每次测试结果都是一样的，可以再给出一些细节吗？

**Comment by ivylinden** [comment]
def compute_saliency_maps( model,testloader):
    model.eval()
    with tqdm(testloader, desc="Validation ", total=len(testloader)) as tepoch:
        running_loss = 0.0
        for inputs, target,_,_ in tepoch:  
            running_loss = 0.0
            imgs = inputs.cuda()
            pids1 = target.cuda()
            # 确保X是需要gradient
            imgs.requires_grad_()
            hrs1= model(imgs,pids1,1)
            loss = torch.abs(hrs1-pids1)
            running_loss = torch.abs(hrs1-pids1).sum()
            running_loss.backward() 
            saliency = imgs.grad.data 
我训练完成后求显著性图，上面是代码，我测试的模型指标没问题，但看每次求得saliency都会变，输入是一维序列，编码值为0,1,2这样的

**Comment by wuhaixu2016** [comment]
您好，您可以通过以下函数严格固定随机数：

def setup_seed(seed):
     torch.manual_seed(seed)
     torch.cuda.manual_seed_all(seed)
     np.random.seed(seed)
     random.seed(seed)
     torch.backends.cudnn.deterministic = True
